File: src/main/java/net/atos/client/zgw/zrc/model/zaakobjecten/ZaakobjectOpenbareRuimte.java
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2025 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.zgw.zrc.model.zaakobjecten;

import java.net.URI;

import nl.info.client.zgw.zrc.model.generated.ObjectTypeEnum;


/**
 * ZaakobjectOpenbareRuimte
 */
public class ZaakobjectOpenbareRuimte extends ZaakobjectMetObjectIdentificatie<ObjectOpenbareRuimte> {

    /**
     * Constructor for JSONB deserialization
     */
    public ZaakobjectOpenbareRuimte() {
    }

    /**
     * Constructor with required attributes
     */
    public ZaakobjectOpenbareRuimte(final URI zaak, final URI bagobjectURI, final ObjectOpenbareRuimte objectOpenbareRuimte) {
        super(zaak, bagobjectURI, ObjectTypeEnum.OPENBARE_RUIMTE, objectOpenbareRuimte);
    }

    @Override
    public String getWaarde() {
        return getObjectIdentificatie().getIdentificatie();
    }
}

File: src/main/kotlin/nl/info/client/zgw/zrc/model/zaakobjecten/ZaakobjectOpenbareRuimte.kt
/*
 * SPDX-FileCopyrightText: 2023, 2025, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.zgw.zrc.model.zaakobjecten

import nl.info.client.zgw.zrc.model.generated.ObjectTypeEnum
import java.net.URI
import java.util.UUID

class ZaakobjectOpenbareRuimte : ZaakobjectMetObjectIdentificatie<ObjectOpenbareRuimte> {
    /**
     * Constructor for JSONB deserialization
     */
    constructor() : super()

    /**
     * Constructor with required attributes
     */
    constructor(
        zaak: URI,
        bagobjectURI: URI?,
        objectOpenbareRuimte: ObjectOpenbareRuimte?,
        url: URI,
        uuid: UUID
    ) : super(zaak, bagobjectURI, ObjectTypeEnum.OPENBARE_RUIMTE, objectOpenbareRuimte, url, uuid)

    override val waarde: String?
        get() = objectIdentificatie?.identificatie
}
